rosbag2_transport/test/rosbag2_transport/test_qos.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <string>
+#include <vector>
+
+#include "rmw/types.h"
+
+#include "qos.hpp"
+
+TEST(TestQoS, serialization)
+{
+  rosbag2_transport::Rosbag2QoS expected_qos;
+  YAML::Node offered_qos_profiles;
+  offered_qos_profiles.push_back(expected_qos);
+
+  std::string serialized = YAML::Dump(offered_qos_profiles);
+  printf("%s\n", serialized.c_str());
```

Comment:
I think `printf` should be removed from tests. If possible, use an EXPECT and pipe in the error message.

Reply:
It wasn't intended to be there, removed.